client/my-sites/domains/domain-management/settings/index.tsx

```diff
@@ -190,6 +202,9 @@ const Settings = ( {
 	};
 
 	const renderContactInformationSecion = () => {
+		if ( ! domain ) {
```

Comment:
With this check in place, we can now remove the `! domain` check on L223 below, as it is now redundant.